18F/identity-idp
# LG-14010 - More detailed error page for Socure errors

Added plumbing and UX display for categorized Socure errors.

changelog Upcoming Features, Socure, Added nice error displays

## 🎫 Ticket

Link to the relevant ticket:
[LG-14010](https://cm-jira.usa.gov/browse/LG-14010)

## 🛠 Summary of changes

Passed the Socure errors through from the original DocV response back through the document capture result, then added a presenter and updated error page to display it all nicely.

## 📜 Testing Plan

This test procedure requires a sandbox set up to use Socure. See jmax
or Doug to coordinate use of one.

- SSM into the sandbox and modify the file
  `/srv/idp/current/app/services/doc_auth/socure/responses/docv_result_response.rb`. Insert
  the following lines after line 36:
```    
    # KLUDGE
    parsed_body['documentVerification']['decision']['value'] = 'reject'
    parsed_body['documentVerification']['reasonCodes'] = ['I848']
```
- Exit the SSM sesion and reboot Puma
- Connect to the IDP, create an account, and begin IdV. Verify that
  you end up on the Category 1 error page.
- Cancel out of IdV and return to your account page.

- SSM into the sandbox and modify the above file once again. Change the error code from 'I848' to 'I849'.
- Exit the SSM session and reboot Puma
- Connect to the IDP, create an account, and begin IdV. Verify that
  you end up on the Category 2 error page.
- Cancel out of IdV and return to your account page.

- SSM into the sandbox and modify the above file once again. Change the error code from 'I849' to 'R827'.
- Exit the SSM session and reboot Puma
- Connect to the IDP, create an account, and begin IdV. Verify that
  you end up on the Category 3 error page.
- Cancel out of IdV and return to your account page.

- SSM into the sandbox and modify the above file once again. Change the error code from 'R827' to 'I808'.
- Exit the SSM session and reboot Puma
- Connect to the IDP, create an account, and begin IdV. Verify that
  you end up on the Category 4 error page.
- Cancel out of IdV and return to your account page.

- SSM into the sandbox and modify the above file once again. Change the error code from 'I808' to 'R845'.
- Exit the SSM session and reboot Puma
- Connect to the IDP, create an account, and begin IdV. Verify that
  you end up on the Category 5 error page.
- Cancel out of IdV and return to your account page.

- SSM into the sandbox and modify the above file once again. Change the error code from 'R845' to 'I856'.
- Exit the SSM session and reboot Puma
- Connect to the IDP, create an account, and begin IdV. Verify that
  you end up on the Category 6 error page.
- Cancel out of IdV and return to your account page.

- SSM into the sandbox and modify the above file once again. This
  time, remove the error code line completely and replace it with
  `raise "Some error"`
- Exit the SSM session and reboot Puma
- Connect to the IDP, create an account, and begin IdV. Verify that
  you end up on the Category 7 error page.

- Cancel out of IdV, return to your account page, and log out.


## 👀 Screenshots

If relevant, include a screenshot or screen capture of the changes.

<details>
<summary>Category 1 error page</summary>

![image](https://github.com/user-attachments/assets/cd37067d-d140-45a4-87c0-972ee62c7e29)

</details>

<details>
<summary>Category 2 error page</summary>

![image](https://github.com/user-attachments/assets/daa6bb71-c0dd-4b62-933f-61571c92a884)</details>

<details>
<summary>Category 3 error page</summary>

![image](https://github.com/user-attachments/assets/ae9369c7-7f45-4e2e-8712-00347ab46d63)

</details>

<details>
<summary>Category 4 error page</summary>

![image](https://github.com/user-attachments/assets/e9fcebeb-e977-4d11-b729-5c5191f2c827)

</details>

<details>
<summary>Category 5 error page</summary>

![image](https://github.com/user-attachments/assets/1f785853-896a-449b-99c3-626a9578335e)

</details>

<details>
<summary>Category 6 errpr page</summary>

![image](https://github.com/user-attachments/assets/facbfeb0-d532-437a-b10d-ba9e0a575ba1)

</details>

<details>
<summary>Category 7 error page</summary>

![image](https://github.com/user-attachments/assets/ce7a2a49-9ce5-4eda-9bbe-b411cd708b7a)

</details>
